src/Command/SplitDb.php

```diff
@@ -0,0 +1,269 @@
+    const TYPE_SALES = 'sales';
+
+    /**
+     * Types and connection for splitting database
+     */
+    const TYPE_MAP = [
+        self::TYPE_QUOTE => [
+            DbConfig::KEY_CONNECTION => DbConfig::CONNECTION_CHECKOUT,
+            DbConfig::KEY_RESOURCE => DbConfig::RESOURCE_CHECKOUT
+        ],
+        self::TYPE_SALES => [
+            DbConfig::KEY_CONNECTION => DbConfig::CONNECTION_SALES,
+            DbConfig::KEY_RESOURCE => DbConfig::RESOURCE_SALES
+        ],
+    ];
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Final database and resource configurations.
+     *
+     * @var DbConfig
+     */
+    private $dbConfig;
+
+    /**
+     * Final configuration for deploy phase
+     *
+     * @var DeployInterface
+     */
+    private $stageConfig;
+
+    /**
+     * Class for configuration merging
+     *
+     * @var ConfigMerger
+     */
+    private $configMerger;
+
+    /**
+     * Factory for creation database configurations
+     *
+     * @var RelationshipConnectionFactory
+     */
+    private $connectionDataFactory;
+
+    /**
+     * Reader for app/etc/env.php file
+     *
+     * @var ConfigReader
+     */
+    private $configReader;
+
+    /**
+     * @var MaintenanceModeSwitcher
+     */
+    private $maintenanceMode;
+
+    /**
+     *  * ./bin/magento shell wrapper.
+     *
+     * @var MagentoShell
+     */
+    private $magentoShell;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param DbConfig $dbConfig
+     * @param DeployInterface $stageConfig
+     * @param ConfigMerger $configMerger
+     * @param RelationshipConnectionFactory $connectionDataFactory
+     * @param ConfigReader $configReader
+     * @param MaintenanceModeSwitcher $maintenanceModeSwitcher
+     * @param MagentoShell $magentoShell
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        DbConfig $dbConfig,
+        DeployInterface $stageConfig,
+        ConfigMerger $configMerger,
+        RelationshipConnectionFactory $connectionDataFactory,
+        ConfigReader $configReader,
+        MaintenanceModeSwitcher $maintenanceModeSwitcher,
+        MagentoShell $magentoShell
+    ) {
+        $this->logger = $logger;
+        $this->dbConfig = $dbConfig;
+        $this->stageConfig = $stageConfig;
+        $this->configMerger = $configMerger;
+        $this->connectionDataFactory = $connectionDataFactory;
+        $this->configReader = $configReader;
+        $this->maintenanceMode = $maintenanceModeSwitcher;
+        $this->magentoShell = $magentoShell;
+        parent::__construct();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $types = array_keys(self::TYPE_MAP);
+        $this->setName(self::NAME)
+            ->setDescription(
+                'Move sales and/or checkout quote related tables to a separate DB servers.'
+                . ' This procedure is not allowed if you do not have prepared DBs'
```

Comment:
`This procedure is not allowed if you do not have prepared DBs` - Are you sure that we need to mention this?
Everyone has prepared DB after the deployment.

Reply:
oh, I got what you wanted to say. Need to rephrase it in some way. Or better to describe the reason of declining in output of the command (after running)